### PR TITLE
chore(vat-data): fix and test defineKind types

### PR DESCRIPTION
**Pull request by turadg**
## Description

Fix problems with `defineKind` that came up while working on https://github.com/Agoric/agoric-sdk/pull/5052 (extracted from that PR)

### Security Considerations

--

### Documentation Considerations

Might need some more documentation. Reviewer requests welcomed.

### Testing Considerations

Improves test coverage

### CODE REVIEW

**Review by mhofman — APPROVED** [review]
This only works for multi-faceted kinds, right?

**Comment by turadg** [comment]
> This only works for multi-faceted kinds, right?

Yep, I'm waiting on the new `defineMultiKind` before fixing that.

**Comment by mhofman** [comment]
Should these be name `Multi` in that case?

**Comment by turadg** [comment]
> Should these be name Multi in that case?

I just pushed a change. I'm inclined to save that for the next PR but if I have to push another change I'll take a pass at naming.

File: yarn.lock [hunk 1]
```diff
@@ -17404,6 +17506,18 @@ tsconfig-paths@^3.12.0:
     minimist "^1.2.0"
     strip-bom "^3.0.0"
 
+tsd@^0.20.0:
+  version "0.20.0"
+  resolved "https://registry.yarnpkg.com/tsd/-/tsd-0.20.0.tgz#0346321ee3c506545486227e488e753109164248"
+  integrity sha512-iba/JlyT3qtnA9t8VrX2Fipu3L31U48oRIf1PNs+lIwQ7n63GTkt9eQlB5bLtfb7nYfy9t8oZzs+K4QEoEIS8Q==
+  dependencies:
+    "@tsd/typescript" "~4.6.3"
+    eslint-formatter-pretty "^4.1.0"
```
Comment (mhofman) [inline yarn.lock:17515]: Ugh why is that package pulling in an `eslint` non-dev dependency?

**Review by mhofman — APPROVED** [review]

**Pull request merged**